flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -2099,13 +2099,13 @@ class ArrayExprLowering {
       }
     }
 
-    // Iteration space is created with outermost columns, innermost rows
+    // Iteration space is created with innermost columns, outermost rows
     fir::DoLoopOp inner;
     auto innerArg = destination.getResult();
     mlir::Value outerRes;
-    llvm::SmallVector<mlir::Value> ivars;
-    auto insPt = builder.saveInsertionPoint();
     const auto loopDepth = loopUppers.size();
+    llvm::SmallVector<mlir::Value> ivars(loopDepth);
+    auto insPt = builder.saveInsertionPoint();
     assert(loopDepth > 0);
     for (auto i : llvm::enumerate(llvm::reverse(loopUppers))) {
```

Comment:
Can't you just remove this call to llvm::reverse here? Other than the comment and the regression test, that ought to be about it.

Reply:
Nah. My bad. This reverse is needed to create the loops "inside out". The vector of indices can be reversed afterwards.